external/libseccomp.cmake

```diff
@@ -5,9 +5,14 @@ ENDIF()
 ADD_LIBRARY(seccomp STATIC IMPORTED)
 
 IF(NOT DEFINED LIBSECCOMP_BUILD_OWN OR LIBSECCOMP_BUILD_OWN STREQUAL "NO")
+    IF(LINK STREQUAL "STATIC")
+        SET(libseccomp_LIB_FILE_NAME "libseccomp.a")
+    ELSE()
+        SET(libseccomp_LIB_FILE_NAME "libseccomp.so")
```

Comment:
To wygląda jak bardzo słaby pomysł, bo *zainstalowany* sio2jail będzie zależał od `libseccomp.so` w katalogu build, który może zniknąć. Poza tym, co jeśli będzie preferował `libseccomp.so` z `/usr/lib`, chociaż tam jest zła wersja?

Reply:
Hmm... jeżeli chodzi o szukanie, to jest ok:

```
teqwve ~/projects/sio2jail/build $ cmake -DLINK=DYNAMIC -DLIBSECCOMP_BUILD_OWN=YES ..
...
-- Libseccomp configuration:
-- - library: /home/teqwve/projects/sio2jail/build/seccomp_project-prefix/lib/libseccomp.so
-- - include directory: /home/teqwve/projects/sio2jail/build/seccomp_project-prefix/include
...
teqwve ~/projects/sio2jail/build $ make VERBOSE=1
/usr/bin/c++  -std=gnu++14 -Wall  -rdynamic CMakeFiles/sio2jail.dir/common/Exception.cc.o (...) CMakeFiles/sio2jail.dir/tracer/Tracee.cc.o  -o sio2jail ../seccomp_project-prefix/lib/libseccomp.so -lrt -lpthread 
...
teqwve ~/projects/sio2jail/build $ ldd ./src/sio2jail 
        linux-vdso.so.1 (0x00007ffea5d23000)
        libseccomp.so.2 => /usr/lib64/libseccomp.so.2 (0x00007f4af8e54000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f4af8c4c000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f4af8a2c000)
        libstdc++.so.6 => /usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/libstdc++.so.6 (0x00007f4af8604000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f4af8274000)
        libgcc_s.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/7.3.0/libgcc_s.so.1 (0x00007f4af805d000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f4af7c95000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f4af94ac000)
teqwve ~/projects/sio2jail/build $ LD_BIND_NOW=1 ./src/sio2jail | head -n5
Error:
    Required argument missing: path

Usage: 
   ./src/sio2jail  [-l <path>] [-B] [-b <string>] ...  [--ustimelimit <time
```

nie ma nawet tej ścieżki w RPATH. Może to różni się od -lseccomp tylko tym, że linker nie musi szukać pliku .so po katalogach?

Jeżeli chodzi o to, że komuś nie będzie pasować, to owszem jest takie ryzyko. Możemy linkować do konkretniejszej wersji (libseccomp.so.2). Zasadniczo tą opcje zrobiłem z myślą o @michcioperz który chciał robić jakieś paczki z s2j, można usunąć jej opis z README, albo dać info, żeby nie robić tego bez wiedzy co to oznacza (IMO, to że STATIC jest domyślną wartością częściowo to rozwiązuje).

Reply:
A stary libseccomp miał skopane tylko headerki czy libkę też?
Może nie pozwalajmy używać statycznego linkowania jeśli `SECCOMP_OWN=YES` ?
No bo to nie ma sensu, żeby budować własnego seccompa, nie instalować go w systemie, a potem w runtimie używać systemowego.
(Chyba że cross-kompilujesz, ale wtedy i tak będziesz to robić inaczej)

Reply:
Hm... ok. Jak budujesz paczkę, też chcesz sprawdzić, czy działa z konkretną wersją libseccomp, którą podasz.